lib/client-api/src/types.ts

```diff
@@ -95,7 +95,7 @@ export interface GetStorybookKind {
 
 // This really belongs in lib/core, but that depends on lib/ui which (dev) depends on app/react
 // which needs this type. So we put it here to avoid the circular dependency problem.
-export type RenderContext = StoreItem & {
+export type RenderContext<StoryFnReturnType = unknown> = StoreItem<StoryFnReturnType> & {
```

Comment:
This required some changes in the other apps besides Vue, but now they all get the correct type from calling `storyFn()` instead of `any`, so it's a win in my opinion. It's technically breaking if there are any other apps out there, that expect `any` instead of `unknown`? `unknown` is generally more type-safe though.